internal/app/orchestrator/orchestrator.go

```diff
@@ -280,10 +280,16 @@ func (o *orchestrator) watchUpstream(
 		case x, more := <-responseChannel:
 			if !more {
 				// A problem occurred fetching the response upstream, log retry.
-				// TODO implement retry/back-off logic on error scenario.
-				// https://github.com/envoyproxy/xds-relay/issues/68
 				o.logger.With("aggregated_key", aggregatedKey).Error(ctx, "upstream error")
 				metrics.OrchestratorWatchErrorsSubscope(o.scope, aggregatedKey).Counter(metrics.ErrorUpstreamFailure).Inc(1)
+
+				// TODO implement retry/back-off logic on error scenario.
+				// https://github.com/envoyproxy/xds-relay/issues/68
```

Comment:
Should we add a comment that also says something like
`Currently we force close all downstream connections, forcing downstream clients to issue a retry.` or we could just add it to the issue too.

Reply:
added it to the issue.